File: backend/src/main/java/com/hongqiao/lims/equipment/Equipment.java
package com.hongqiao.lims.equipment;

import com.hongqiao.lims.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import java.time.LocalDate;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "equipment")
public class Equipment extends BaseEntity {

    @Column(name = "asset_no", nullable = false, unique = true, length = 64)
    private String assetNo;

    @Column(nullable = false, length = 128)
    private String name;

    @Column(length = 64)
    private String category;

    @Column(length = 128)
    private String model;

    @Column(length = 128)
    private String manufacturer;

    @Column(name = "serial_no", length = 128)
    private String serialNo;

    @Column(length = 128)
    private String location;

    @Column(nullable = false, length = 32)
    private String status = "idle";

    @Column(name = "purchase_date")
    private LocalDate purchaseDate;

    @Column(name = "calibration_date")
    private LocalDate calibrationDate;

    @Column(name = "calibration_due")
    private LocalDate calibrationDue;

    @Column(name = "calibration_cycle_days")
    private Integer calibrationCycleDays;

    @Column(columnDefinition = "text")
    private String remark;
}

File: backend/app/schemas/equipment.py
from datetime import date

from pydantic import BaseModel


class EquipmentBase(BaseModel):
    asset_no: str
    name: str
    category: str | None = None
    model: str | None = None
    manufacturer: str | None = None
    serial_no: str | None = None
    location: str | None = None
    status: str = "idle"
    purchase_date: date | None = None
    calibration_date: date | None = None
    calibration_due: date | None = None
    calibration_cycle_days: int | None = None
    remark: str | None = None


class EquipmentCreate(EquipmentBase):
    pass


class EquipmentUpdate(BaseModel):
    name: str | None = None
    category: str | None = None
    model: str | None = None
    manufacturer: str | None = None
    serial_no: str | None = None
    location: str | None = None
    status: str | None = None
    purchase_date: date | None = None
    calibration_date: date | None = None
    calibration_due: date | None = None
    calibration_cycle_days: int | None = None
    remark: str | None = None


class EquipmentOut(EquipmentBase):
    id: int

    model_config = {"from_attributes": True}
